src/resources/transactions/firstTransactionTimestampQuery.ts

```diff
@@ -0,0 +1,104 @@
+import { useQuery } from '@tanstack/react-query';
+import PQueue from 'p-queue/dist';
+
+import { fetchENSAddress } from '@/hooks/useENSAddress';
+import {
+  createQueryKey,
+  queryClient,
+  QueryConfig,
+  QueryFunctionArgs,
+  QueryFunctionResult,
+} from '@/react-query';
+import { getFirstTransactionTimestamp } from '@/utils/ethereumUtils';
+
+// ///////////////////////////////////////////////
+// Query Types
+
+export type FirstTransactionTimestampArgs = {
+  addressOrName?: string;
+};
+
+// ///////////////////////////////////////////////
+// Query Key
+
+export const firstTransactionTimestampQueryKey = ({
+  addressOrName,
+}: FirstTransactionTimestampArgs) =>
+  createQueryKey(
+    'firstTransactionTimestamp',
+    { addressOrName },
+    { persisterVersion: 1 }
+  );
+
+export type FirstTransactionTimestampQueryKey = ReturnType<
+  typeof firstTransactionTimestampQueryKey
+>;
+
+// ///////////////////////////////////////////////
+// Query Function
+
+const queue = new PQueue({ interval: 1000, intervalCap: 5 });
+
+export async function firstTransactionTimestampQueryFunction({
+  queryKey: [{ addressOrName }],
+}: QueryFunctionArgs<typeof firstTransactionTimestampQueryKey>) {
+  if (!addressOrName) return null;
```

Comment:
Nit: is there a reason to return `null` over `undefined`? Sometimes plays nicer with TS I think, but not a huge deal.

Reply:
Unfortunately, React Query doesn't support returning `undefined` as it is not serializable :(

Reply:
Works for me!